server/etcdserver/server.go

```diff
@@ -1192,7 +1194,9 @@ func (s *EtcdServer) applyAll(ep *etcdProgress, apply *apply) {
 	// storage, since the raft routine might be slower than apply routine.
 	<-apply.notifyc
 
-	s.triggerSnapshot(ep)
+	if ep.appliedi-ep.snapi > s.Cfg.SnapshotCount {
+		s.triggerSnapshot(ep)
```

Comment:
Please add a test that makes sure the snapshot gets executed (e.g. e2e test, that greps logs). 

E.g. GoAttach() is used by triggerSnapshot and 'if server is stopping' it skips execution of goroutines.

Reply:
Done. Verified in 100 runs that test passes.